syntax= "proto3";

// Details on what these are can be found at https://grpc.io/docs/guides/concepts.html
// 
service QuestionService {
	rpc UnaryRequest(QuestionRequest) returns (QuestionResponse) {}
	rpc CountWordsInFile(stream FileStream) returns (stream WordCount) {} 
}

message QuestionRequest {
	string query = 1;
}

message QuestionResponse {
	string reply = 1;
}

message FileData {
	string file_data = 1;
}

message WordCount { 
	int32 count = 1;
}